symengine/tests/basic/test_basic.cpp

```diff
@@ -383,6 +383,13 @@ TEST_CASE("compare: Basic", "[basic]")
     r2 = mul(x, z);
     CHECK(r1->compare(*r2) == -1);
     CHECK(r2->compare(*r1) == 1);
+//    CHECK(r1->compare(*r2) == -1);
+//    CHECK(r2->compare(*r1) == 1);
+
+    r1 = mul(y, x);
+    r2 = mul(x, z);
+//    CHECK(r1->compare(*r2) == -1);
+//    CHECK(r2->compare(*r1) == 1);
```

Comment:
These tests actually fail. I need to debug it. For some reason, the `compare` returns the `+1` and `-1` the other way. This might indicate a bug in our code, that never showed up, I need to see if this comparisons is platform dependent or not.